app/controllers/users_controller.rb

```diff
@@ -169,6 +169,10 @@ def onboarding_update
     user_params = { saw_onboarding: true }
 
     if params[:user]
+      if params[:user].key?(:username) && params[:user][:username].blank?
```

Comment:
Ruby added the `dig` method a while back to make checks like this one easier:

```suggestion
      if params.dig(:user, :username).blank?
```

Reply:
I had to tweak a test because there was a case in which we sent:  `{ user: { last_onboarding_page: "v2: personal info form" } }` causing this check to raise an error. However, I don't think that test reflects the way this form is going to be used and we should always expect a username so that's fine.